tutorial/numpy-functions.md

```diff
@@ -0,0 +1,385 @@
+# Supported Numpy function by Pyccel
+
+In Pyccel we try to support the Numpy functions which developers use the most.. Here are some of them:
+
+## [Norm](https://numpy.org/doc/stable/reference/generated/numpy.linalg.norm.html)
+
+-   Supported parameters:
+
+    x: array_like
+       Input array. If axis is None, x must be 1-D or 2-D, unless ord is None.
+       If both axis and ord are None, the 2-norm of x.ravel will be returned.
+
+    axis: {None, int, 2-tuple of ints}, optional.
+         If axis is an integer, it specifies the axis of x along which to compute the vector norms.
+         If axis is a 2-tuple, it specifies the axes that hold 2-D matrices, and the matrix norms of
+         these matrices are computed. If axis is None then either a vector norm (when x is 1-D) or a
+         matrix norm (when x is 2-D) is returned. The default is None. New in version 1.8.0.
+
+-   Supported languages: Fortran (2-norm)
+
+-   python code:
+
+    ```python
+    from numpy.linalg import norm
+    from numpy import array
+    arr1 = array([1,2,3,4])
+    nrm = norm(arr1)
+    print(nrm)
+
+    arr2 = array([[1,2,3,4],[4,3,2,1]])
+    nrm2 = norm(arr2, axis=1)
+    print(nrm)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_boo
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr1(:)
+    real(C_DOUBLE) :: nrm
+    integer(C_INT64_T), allocatable :: arr2(:,:)
+    real(C_DOUBLE), allocatable :: nrm2(:)
+
+    allocate(arr1(0:3_C_INT64_T))
+    arr1 = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    nrm = Norm2(Real(arr1, C_DOUBLE))
+    print *, nrm
+    allocate(arr2(0:3_C_INT64_T, 0:1_C_INT64_T))
+    arr2 = reshape([[1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T], [ &
+      4_C_INT64_T, 3_C_INT64_T, 2_C_INT64_T, 1_C_INT64_T]], [ &
+      4_C_INT64_T, 2_C_INT64_T])
+    allocate(nrm2(0:1_C_INT64_T))
+    nrm2 = Norm2(Real(arr2, C_DOUBLE),2_C_INT64_T - 1_C_INT64_T)
```

Comment:
:( this didn't neaten up like the rest. Can you add a `simplify = True` to this `PyccelMinus` please

Reply:
Fixed with commit b537680